uiTests/build.gradle

```diff
@@ -0,0 +1,58 @@
+apply plugin: 'org.gradle.scala'
+
+import org.gradle.api.tasks.testing.logging.TestLogEvent
+
+List<String> uiTestTargets = project.properties.against \
+  ? resolveIntelliJVersions(project.properties.against)
+  : [intellijVersions.buildTarget]
+
+uiTestTargets.each { version ->
+  tasks.register("uiTest_${version}", Test) {
+    description = 'Runs UI tests.'
+    group = 'verification'
+
+    testClassesDirs = sourceSets.uiTest.output.classesDirs
+    classpath = configurations.uiTestRuntimeClasspath + sourceSets.uiTest.output
+
+    def buildPlugin = tasks.getByPath(':buildPlugin')
+    dependsOn buildPlugin
+
+    systemProperty 'ui-test.plugin.path', buildPlugin.outputs.files[0].path
+    systemProperty 'ui-test.intellij.version', version
+
+    if (!isCI) {
+      outputs.upToDateWhen { false }
+    }
+
+    if (project.properties.tests) {
+      filter {
+        includeTestsMatching "*.*${project.properties.tests}*"
+      }
+    }
+
+    if (project.hasProperty('headless')) {
+      environment 'IDEPROBE_DISPLAY', 'xvfb'
+      // TODO (VirtusLab/ide-probe#195): remove the need to set these env variables
+      environment 'IDEPROBE_TEST_SUITE', 'uiTest' + version
+      environment 'IDEPROBE_TEST_CASE', 'artifacts'
+    }
+
+    testLogging {
+      events += [TestLogEvent.STANDARD_OUT, TestLogEvent.STANDARD_ERROR]
+    }
+  }
+}
+
+task uiTest {
+  dependsOn tasks.matching { Task task -> task.name.startsWith("uiTest_") }
+}
+
+dependencies {
+  // Needed so that the main classpath of the plugin is considered by Gradle as an input to `uiTest` task.
+  // Otherwise, changes in the plugin would NOT cause the UI tests to be re-executed when caching is enabled.
```

Comment:
This is the new addition. Previously, UI tests resided in root project, so they depended on `src/main` code of root project anyway. Now they're in a **sub**project... so I needed to add this atypical construct (**sub**project depending on root project, not the other way round)

Reply:
This entire extraction to a separate subproject was needed as fix (workaround, kinda, again...) for #945